### DISCUSSION TITLE
23.01.2024

### DISCUSSION THREAD

**Discussion by Roadraudi** [Roadraudi/Codespace_Repository#12] [Announcements]
Hallo liebe Leser,

ich hoffe, diese Woche haben euch meine zahlreichen Code-Zeilen gefallen. In meinem heutigen Weekly Coding Update möchte ich auf meine beiden Programme eingehen, die beide etwas mit Passwörtern zu tun haben.

Der erste [Passwort-Prüfer](https://github.com/Roadraudi/Codespace_Repository/blob/codespace-upgraded-space-trout-r4466vjqx6wgf5vvv/src/PasswortPruefer.py) untersucht ein zufällig eingegebenes Passwort auf Dopplungen. Wenn das Passwort keine Dopplungen enthält, ist es sicher. Andernfalls wird ein Wert, der doppelt ist, durch einen neuen Buchstaben, eine Zahl oder ein Sonderzeichen ersetzt, das noch nicht im Passwort enthalten ist. Das neue mögliche Passwort wird dem Nutzer zurückgegeben.

Der zweite [Passwortprüfer](https://github.com/Roadraudi/Codespace_Repository/blob/codespace-upgraded-space-trout-r4466vjqx6wgf5vvv/src/PasswortPruefer2.py) basiert mehr auf dem Prinzip, des heutigen Passwort Prüfen wie auf Webseiten wenn man zum Beispiel einen neuen Account erstellen möchte. Dieser prüft das vom Nutzer eingegebene Passwort auf die Länge (mindestens 8 Zeichen) und ob Groß- und Kleinbuchstaben bzw. eine Zahl enthalten ist. Auf dieser Grundlage bestimmt er, ob das Passwort sicher ist oder nicht.

Außerdem habe ich begonnen, [Tic-Tac-Toe](https://github.com/Roadraudi/Codespace_Repository/blob/codespace-upgraded-space-trout-r4466vjqx6wgf5vvv/src/Tic-Tac-Toe.py) zu programmieren. Bisher steht nur das Spielfeld, aber die Gameengine möchte ich in dieser Woche noch hinzufügen. Als erste Lösung soll das Programm nur in der Kommandozeile funktionieren, aber ich überlege, dies vielleicht auch später noch mit einer GUI zu programmieren.

Dabei kommen wir auch zu meinem letzten [Programm](https://github.com/Roadraudi/Codespace_Repository/blob/codespace-upgraded-space-trout-r4466vjqx6wgf5vvv/src/Pygame%20Test.py), das ich heute hochgeladen habe, bei dem ich mich das erste Mal mit Pygame auseinandergesetzt habe. Dazu habe ich mich auf der folgenden [Internetseite](https://pypi.org/project/pygame/) näher damit informiert zum download sowie Youtube-Tutorials angeschaut und nachprogrammiert. Die folgenden beiden Videos könnten vielleicht euch auch weiterhelfen wenn ihr noch blutige Anfänger seid in Pygame: [Tutorial auf Deutsch](https://www.youtube.com/watch?v=6ytwfT3brSc&list=PLjuHXsDGkbTSpTMJ4MXTiQXN4AeOzpu1w) oder [Tutorial auf Englisch](https://www.youtube.com/watch?v=y9VG3Pztok8&t=360s). Bei meinem Programm erscheint ein Fenster mit einem Rechteck das man mit den Pfeil Tasten lenken kann, meiner Meinung nach ein einfaches Programm aber sehr gut um sich mit Pygame vertraut zu machen.

Also Ausblick für die Zukunft steht einmal Tic-Tac-Toe fertig programmieren sowie sich mit Pygame noch näher zu beschäftigen. Außerdem möchte ich mich noch mit weiteren librarys beschäftigen wie numpy, panda und so weiter.

**Comment by dhbwJK23** [comment]
Der Beginn einer großen Karriere als Entwickler erstklassiger Spiele?
So oder so ne gute Lerngelegenheit um etwas zu lernen was nicht unbedingt jeder kann.
Weiter so!

**Comment by carolina-cza** [comment]
Sehr cool.Pygame ist ein echt nützliches und cooles Tool
